src/me/remie/vulcan/leaguetasks/utils/LeaguePanel.java

```diff
@@ -0,0 +1,119 @@
+                final SimpleWidget selectedTierOption = typeDropDownMenu.getChild(4);
+                final SimpleWidget dropDownMenu = ctx.widgets.getWidget(657, 36);
+                if (!chooseOptionInDropdown(selectedTierOption, dropDownMenu,1)) {
+                    ctx.log("Something went wrong choosing type option");
+                    return false;
+                }
+            }
+        }
+
+        final SimpleWidget areaDropDownMenu = ctx.widgets.getWidget(657,29);
+        if (areaDropDownMenu != null) {
+            if (!isOptionSelected(areaDropDownMenu, 4, "all")) {
+                final SimpleWidget selectedTierOption = areaDropDownMenu.getChild(4);
+                final SimpleWidget dropDownMenu = ctx.widgets.getWidget(657, 37);
+                if (!chooseOptionInDropdown(selectedTierOption, dropDownMenu,1)) {
+                    ctx.log("Something went wrong choosing area option");
+                    return false;
+                }
+            }
+        }
+
+        final SimpleWidget completedDropDownMenu = ctx.widgets.getWidget(657,31);
+        if (completedDropDownMenu != null) {
+            if (!isOptionSelected(completedDropDownMenu, 4, "all")) {
+                final SimpleWidget selectedTierOption = completedDropDownMenu.getChild(4);
+                final SimpleWidget dropDownMenu = ctx.widgets.getWidget(657, 39);
+                if (!chooseOptionInDropdown(selectedTierOption, dropDownMenu,1)) {
+                    ctx.log("Something went wrong choosing completed option");
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    private boolean chooseOptionInDropdown(SimpleWidget dropDownWidget, SimpleWidget dropDownOptionList, int option) {
+        if (dropDownWidget == null || dropDownWidget.getChild(option) == null) return false;
+        dropDownWidget.click(1);
+        ctx.sleep(1300);
+        if (dropDownOptionList != null && !dropDownOptionList.isHidden()) {
+            final SimpleWidget selectedOption = dropDownOptionList.getChild(option);
+            if (selectedOption != null) {
+                selectedOption.click(1);
+                ctx.sleep(1300);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isLeaguePanelOpen() {
+        final SimpleWidget leaguePanelWidget = ctx.widgets.getWidget(657, 18);
+        return leaguePanelWidget != null && !leaguePanelWidget.isHidden();
+    }
+
+    public boolean openLeaguePanel() {
+        if (isLeaguePanelOpen()) return true;
+        ctx.log("Opening leagues menu");
+        ctx.game.tab(Game.Tab.QUESTS);
+        ctx.menuActions.clickButton(41222167);
+        ctx.sleep(1000);
+        ctx.menuActions.clickButton(42991640);
+        return ctx.onCondition(() -> isLeaguePanelOpen(), 1000, 5);
+    }
+
+    public SimpleWidget getLeaguePanelWidget() {
+        return ctx.widgets.getWidget(657,18);
```

Comment:
```suggestion
        return ctx.widgets.getWidget(WIDGET_ID, 18);
```

Let's create a field at the top of the class to store the widget id, instead of calling it in every single function.
`private static final int WIDGET_ID = 657;`